controlplane/eks/api/v1beta2/awsmanagedcontrolplane_webhook.go

```diff
@@ -392,6 +394,22 @@ func (r *AWSManagedControlPlane) validateDisableVPCCNI() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSManagedControlPlane) validateFilterPrivateSubnets() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.FilterPrivateSubnets {
```

Comment:
I don't think we should do this check here. The subnet details are populated during reconciliation (if you don't explicitly set them) and so there will be no private subnets. We could do this check as part of the reconciliation instead,

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/network/subnets.go#L81

Reply:
>The subnet details are populated during reconciliation (if you don't explicitly set them) and so there will be no private subnets

@richardcase  - we do specify the subnet details explicitly when we create the `awsmanagedcontrolplane` resource. Would I still need to move the validation to [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/network/subnets.go#L81)